File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03561/p03561_num6_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int K, N;
    cin >> K >> N;

    // Calculate X, the total number of sequences listed in FEIS
    long long X = 1;
    for (int i = 0; i < N; i++) {
        X *= K;
    }

    // Calculate the (X/2)-th smallest sequence
    vector<int> sequence(N);
    long long middle = (X + 1) / 2; // Rounded up to the nearest integer

    for (int i = 0; i < N; i++) {
        sequence[i] = (middle - 1) % K + 1;
        middle = (middle - 1) / K;
    }

    // Print the sequence
    for (int i = 0; i < N; i++) {
        cout << sequence[i] << " ";
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03561/p03561_num6_parsed.py
# Python program to find the (X/2)-th lexicographically smallest sequence

def find_lexicographically_smallest_sequence(K, N):
    # Calculate the total number of sequences
    X = K ** N

    # Calculate the (X/2)-th (rounded up) lexicographically smallest sequence
    sequence = []
    for i in range(N-1, -1, -1):
        quotient = X // K
        remainder = X % K

        if remainder == 0:
            sequence.append(K)
            X = quotient
        else:
            sequence.append(remainder)
            X = quotient + 1

    # Print the sequence
    print(*sequence)

# Read the input values
K, N = map(int, input().split())

# Find and print the lexicographically smallest sequence
find_lexicographically_smallest_sequence(K, N)